drivers/sht2x/sht2x.c

```diff
@@ -0,0 +1,476 @@
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_sht2x
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the SHT2x temperature and
+ *              humidity sensor.
+ *
+ * @author      Kees Bakker <kees@sodaq.com>
+ * @author      George Psimenos <gp7g14@soton.ac.uk>
+ * @author      Steffen Robertz <steffen.robertz@rwth-aachen.de>
+ *
+ * @}
+ */
+
+#include <math.h>
+
+#include "log.h"
+#include "sht2x.h"
+#include "sht2x_params.h"
+#include "periph/i2c.h"
+#include "xtimer.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+/**
+ * @brief       The number of retries when doing a polled measurement
+ */
+#define MAX_RETRIES         20
+
+/**
+ * @brief       A few helper macros
+ */
+#define BUS                 (dev->params.i2c_dev)
+#define ADDR                (dev->params.i2c_addr)
+#define MODE                (dev->params.measure_mode)
+#define CRC                 (dev->params.is_crc_enabled)
+
+typedef enum {
+    eTempHoldCmd        = 0xE3,     /**< trigger temp measurement, hold master */
+    eRHumidityHoldCmd   = 0xE5,     /**< trigger humidity measurement, hold master */
+    eWriteUserCmd       = 0xE6,     /**< write user register */
+    eReadUserCmd        = 0xE7,     /**< read user register */
+    eTempNoHoldCmd      = 0xF3,     /**< trigger temp measurement, no hold master (poll) */
+    eRHumidityNoHoldCmd = 0xF5,     /**< trigger humidity measurement, no hold master (poll) */
+    eIdentifierCmd      = 0xFA,     /**< read identifier */
+    eSoftResetCmd       = 0xFE,     /**< soft reset */
+} cmd_t;
+
+typedef enum {
+    SHT2X_MEASURE_TEMP,
+    SHT2X_MEASURE_RH,
+} measure_type_t;
+
+/**
+ * @brief   Register addresses to read SHT2x Identification Code.
+ */
+static const uint16_t first_mem_addr = 0x0FFA;
+static const uint16_t second_mem_addr = 0xC9FC;
+
+static int read_sensor(const sht2x_t* dev, cmd_t command, uint16_t *val);
+static int read_sensor_poll(const sht2x_t* dev, cmd_t command, uint16_t *val);
+static uint8_t sht2x_checkcrc(uint8_t data[], uint8_t nbrOfBytes, uint8_t checksum);
+static void sleep_during_temp_measurement(sht2x_res_t res);
+static void sleep_during_hum_measurement(sht2x_res_t resolution);
+
+/*---------------------------------------------------------------------------*
+ *                          SHT2x Core API                                   *
+ *---------------------------------------------------------------------------*/
+
+int sht2x_init(sht2x_t* dev, const sht2x_params_t* params)
+{
+    int i2c_result;
+
+    dev->params = *params;
+    if (dev->params.resolution != SHT2X_RES_12_14BIT &&
+        dev->params.resolution != SHT2X_RES_8_12BIT &&
+        dev->params.resolution != SHT2X_RES_10_13BIT &&
+        dev->params.resolution != SHT2X_RES_11_11BIT) {
+        return SHT2X_ERR_OTHER;
```

Comment:
That's not documented in the header.

Reply:
OK I'll add some more error codes and use these